include/npcomp/Dialect/ATen/GeneratedATenBuiltinOps.td

```diff
@@ -2021,3 +2263,230 @@ def ATen_MaxPool1dOp: ATen_ImmutableTensorOp<"max_pool1d", [NoSideEffect]> {
+    about not in-place updating of such a result should be treated as UB
+    when compiled.
+  }];
+}
+
+def ATen_ExpandOp: ATen_ImmutableTensorOp<"expand", [NoSideEffect]> {
+  let summary = "expand(*sizes) -> Tensor";
+  let arguments = (ins
+    ATen_AnyTensor:$input,
+    ATen_AnyScalar:$sizes,
+    ATen_BoolScalar:$implicit
+  );
+  let results = (outs
+    ATen_AnyTensor:$result
+  );
+  let description = [{
+    Returns a new view of the :attr:`self` tensor with singleton dimensions expanded
+    to a larger size.
+
+    Passing -1 as the size for a dimension means not changing the size of
+    that dimension.
+
+    Tensor can be also expanded to a larger number of dimensions, and the
+    new ones will be appended at the front. For the new dimensions, the
+    size cannot be set to -1.
+
+    Expanding a tensor does not allocate new memory, but only creates a
+    new view on the existing tensor where a dimension of size one is
+    expanded to a larger size by setting the ``stride`` to 0. Any dimension
+    of size 1 can be expanded to an arbitrary value without allocating new
+    memory.
+
+    Args:
+        *sizes (torch.Size or int...): the desired expanded size
+
+    .. warning::
+
+        More than one element of an expanded tensor may refer to a single
+        memory location. As a result, in-place operations (especially ones that
+        are vectorized) may result in incorrect behavior. If you need to write
+        to the tensors, please clone them first.
+
+    Example::
+
+        >>> x = torch.tensor([[1], [2], [3]])
+        >>> x.size()
+        torch.Size([3, 1])
+        >>> x.expand(3, 4)
+        tensor([[ 1,  1,  1,  1],
+                [ 2,  2,  2,  2],
+                [ 3,  3,  3,  3]])
+        >>> x.expand(-1, 4)   # -1 means not changing the size of that dimension
+        tensor([[ 1,  1,  1,  1],
+                [ 2,  2,  2,  2],
+                [ 3,  3,  3,  3]])
+
+    MLIR Specific Notes
+    -------------------
+    See notes for the 'as_strided' op.
+  }];
+}
+
+def ATen_SqueezeOp: ATen_ImmutableTensorOp<"squeeze", [NoSideEffect]> {
+  let summary = "squeeze(input, dim=None, out=None) -> Tensor";
+  let arguments = (ins
+    ATen_AnyTensor:$input,
+    ATen_IntScalar:$dim
+  );
+  let results = (outs
+    ATen_AnyTensor:$result
+  );
+  let description = [{
+    Returns a tensor with all the dimensions of :attr:`input` of size `1` removed.
+
+    For example, if `input` is of shape:
+    :math:`(A \times 1 \times B \times C \times 1 \times D)` then the `out` tensor
+    will be of shape: :math:`(A \times B \times C \times D)`.
+
+    When :attr:`dim` is given, a squeeze operation is done only in the given
+    dimension. If `input` is of shape: :math:`(A \times 1 \times B)`,
+    ``squeeze(input, 0)`` leaves the tensor unchanged, but ``squeeze(input, 1)``
+    will squeeze the tensor to the shape :math:`(A \times B)`.
+
+    .. note:: The returned tensor shares the storage with the input tensor,
```

Comment:
We don't seem to be modeling this part correctly? Same with other view-returning ops.

Reply:
Maybe add a TODO/ "mlir-specific warning" like you have for some of the others.

Reply:
Yes, and I think I left a note in there about that but will strengthen it. I've got a fairly concrete way of modeling these crystalizing in my mind, but it needs a design written up. It has been very useful for me to just see the inventory and forms so far, so I'm inclined to do that for the other oddball patterns in the op set, even if modeled wrong for now, because it will make it easier to reason about the design to at least have something to refer to.